deep_qa/common/params.py

```diff
@@ -118,15 +141,21 @@ def log_recursively(parameters, history):
         log_recursively(self.params, self.history)
         return self.params
 
-    def __check_is_dict(self, new_history, value):
-        if isinstance(value, dict):
-            new_history = new_history if self.history == "" else self.history + "." + new_history
-            return Params(value, new_history)
-        else:
-            return value
+    def assert_empty(self, class_name: str):
+        """
+        Raises a ``ConfigurationError`` if ``self.params`` is not empty.  We take ``class_name`` as
+        an argument so that the error message gives some idea of where an error happened, if there
+        was one.  ``class_name`` should be the name of the `calling` class, the one that got extra
+        parameters (if there are any).
+        """
+        if len(self.params) != 0:
+            raise ConfigurationError("Extra parameters passed to {}: {}".format(class_name, self.params))
 
     def __getitem__(self, key):
-        return self.__check_is_dict(key, self.params[key])
+        if key in self.params:
```

Comment:
Because it's actually a pyhocon object, it raises a different error, which breaks the typical dictionary API.  Confusing and unfortunate...